File: CS/Controllers/WebDocumentViewerWebApiController.cs
using System.Web.Http;
using DevExpress.Web.Mvc;
using DevExpress.XtraReports.Web.Native.ClientControls;
using DevExpress.XtraReports.Web.WebDocumentViewer;
using DevExpress.XtraReports.Web.WebDocumentViewer.Native.Services;

namespace MVCViewerWidgetDemo.Controllers {
    public class WebDocumentViewerWebApiController : ApiController {
        readonly ManagerModuleSubscriber<IWebDocumentViewerRequestManager> requestProcessor;

        public WebDocumentViewerWebApiController() {
            requestProcessor = new ManagerModuleSubscriber<IWebDocumentViewerRequestManager>(DefaultWebDocumentViewerContainer.Current, "");
        }

        public void Get() {
            System.Web.HttpContext.Current.Response.AppendHeader("Access-Control-Allow-Origin", "*");
            requestProcessor.ProcessRequest();
        }

        public void Post() {
            System.Web.HttpContext.Current.Response.AppendHeader("Access-Control-Allow-Origin", "*");
            requestProcessor.ProcessRequest();
        }
    }
}

File: VB/Controllers/WebDocumentViewerWebApiController.vb
Imports System.Web.Http
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraReports.Web.Native.ClientControls
Imports DevExpress.XtraReports.Web.WebDocumentViewer
Imports DevExpress.XtraReports.Web.WebDocumentViewer.Native.Services

Namespace MVCViewerWidgetDemo.Controllers
    Public Class WebDocumentViewerWebApiController
        Inherits ApiController

        Private ReadOnly requestProcessor As ManagerModuleSubscriber(Of IWebDocumentViewerRequestManager)

        Public Sub New()
            requestProcessor = New ManagerModuleSubscriber(Of IWebDocumentViewerRequestManager)(DefaultWebDocumentViewerContainer.Current, "")
        End Sub

        Public Sub [Get]()
            System.Web.HttpContext.Current.Response.AppendHeader("Access-Control-Allow-Origin", "*")
            requestProcessor.ProcessRequest()
        End Sub

        Public Sub Post()
            System.Web.HttpContext.Current.Response.AppendHeader("Access-Control-Allow-Origin", "*")
            requestProcessor.ProcessRequest()
        End Sub
    End Class
End Namespace